src/coreclr/jit/lsra.cpp

```diff
@@ -5076,6 +5077,11 @@ void LinearScan::allocateRegisters()
                 {
                     allocate = false;
                 }
+                else if (lclVarInterval->registerType == TYP_SIMD64)
+                {
+                    allocate                           = false;
+                    lclVarInterval->isPartiallySpilled = true;
```

Comment:
I thought we were spilling the entire Vector512 since we can't efficiently spill just the upper 384 bits. Should this be setting `isPartiallySpilled`?

Reply:
I suggested to set `isPartiallySpilled` in https://github.com/dotnet/runtime/pull/80960#discussion_r1107529670 instead of introducing a new variable. This will be still spilled entirely to the memory, but we will just reuse `isPartiallySpilled` because the underlying concept is still same where we would save partial (or full, in case of ZMM) and then restore the partial (or full, in case of ZMM).

Reply:
Wouldn't it be considered "fully saved" if this wasn't set? Not sure I understand the benefit of setting it in this scenario?

Reply:
The semantics of `isPartiallySpilled` is you set it during `RefTypeUpperVectorSave` and when you are allocating for `RefTypeUpperVectorRestore` you would restore it only if it was set. So essentially, this is just a way to say that the interval was spilled and needs to restore. When we restore, we need to allocate register to restore the variable from stack to memory, so `isPartiallySpilled` tracks it.